tests/test_project_files.py

```diff
@@ -81,141 +81,165 @@ def find_current_browser(driver):
 
-        file_name = 'rename_' + find_current_browser(driver) + '_' + provider + '.txt'
-        new_file, metadata = osf_api.upload_fake_file(session=session, node=node, name=file_name, provider=provider)
+            files_page = FilesPage(driver, guid=node_id)
+            files_page.goto()
 
-        files_page = FilesPage(driver, guid=node_id)
-        files_page.goto()
+            row = find_row_by_name(driver, provider, new_file)
+            row.click()
+            rename_button = find_toolbar_button_by_name(driver, 'Rename')
+            rename_button.click()
+            rename_text_box = driver.find_element_by_id('renameInput')
 
-        row = find_row_by_name(driver, provider, new_file)
-        row.click()
-        rename_button = find_toolbar_button_by_name(driver, 'Rename')
-        rename_button.click()
-        rename_text_box = driver.find_element_by_id('renameInput')
+            for _ in range(len(new_file)):
+                rename_text_box.send_keys(Keys.BACKSPACE)
 
-        for _ in range(len(new_file)):
-            rename_text_box.send_keys(Keys.BACKSPACE)
+            new_name = find_current_browser(driver) + '_' + provider + '_renamed.txt'
+            rename_text_box.send_keys(new_name)
+            rename_text_box.send_keys(Keys.RETURN)
 
-        new_name = find_current_browser(driver) + '_' + provider + '_renamed.txt'
-        rename_text_box.send_keys(new_name)
-        rename_text_box.send_keys(Keys.RETURN)
+            # Wait for 5 seconds for Rename message to show
+            WebDriverWait(driver, 5).until(EC.visibility_of_element_located((By.CLASS_NAME, 'text-muted')))
+            # Wait a maximum of 20 seconds for Rename message to resolve
+            WebDriverWait(driver, 20).until(EC.invisibility_of_element_located((By.CLASS_NAME, 'text-muted')))
 
-        # Wait for 5 seconds for Rename message to show
-        WebDriverWait(driver, 5).until(EC.visibility_of_element_located((By.CLASS_NAME, 'text-muted')))
-        # Wait a maximum of 20 seconds for Rename message to resolve
-        WebDriverWait(driver, 20).until(EC.invisibility_of_element_located((By.CLASS_NAME, 'text-muted')))
+            files_page.goto()
+            # Test old file name does not exist
+            old_file = find_row_by_name(driver, provider, file_name)
+            assert old_file is None
 
-        files_page.goto()
-        # Test old file name does not exist
-        old_file = find_row_by_name(driver, provider, file_name)
-        assert old_file is None
+            # Test that new file name is present and visible
+            renamed_file = find_row_by_name(driver, provider, new_name)
+            assert new_name in renamed_file.text
 
-        # Test that new file name is present and visible
-        renamed_file = find_row_by_name(driver, provider, new_name)
-        assert new_name in renamed_file.text
+        except FileNotFoundError:
```

Comment:
The pythonic way is to try as less code as possible and have multiple tries to catch different exceptions. However, given the nature of selenium tests, which is the same reason why neither setup / tear-up nor decorators work, we can relax it to cater for our needs. As discussed:

* Don't try the set-up code
* We still have to put assertion in the try though due to the clean-up limitation
* Catch the right exception, for now I think it is selenium's timeout exception` and Python's assertion exception
* Re-raise the exception after cleaning up (no need to handle the two exceptions separately

Apply this to all touched tests :)

Reply:
It looks like we should use a Try/Finally here. Originally I wanted to catch all exceptions and re-throw them so I we just took out the except clause.